src/bp/core/routers/admin/bots.ts

```diff
@@ -53,8 +56,24 @@ export class BotsRouter implements CustomRouter {
 
         this.workspaceService.assertUserExists(req.tokenUser.email)
 
-        await this.botService.addBot(bot, req.body.template)
-        await this.workspaceService.addBotRef(bot.id)
+        const botExists = (await this.botService.listAvailableBots()).includes(bot.id)
+        const botLinked = (await this.workspaceService.getBotRefs()).includes(bot.id)
+
+        if (botExists && botLinked) {
+          throw new ConflictError(`Bot ${bot.id} already exists and is already linked in workspace`)
+        }
+
+        if (botExists) {
+          this.logger.warn(`Bot ${bot.id} already exists. Linking to workspace`)
```

Comment:
`Linking to workspace` sounds like you're linking it for him. Don't think its the case